File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    words = list(s)
    skip_words = list(skip)
    
    # print(chr(ord(words[0]) + 1))
    
    answer = ''
    
    for word in words:
        tmp_index = index
        while(tmp_index):
            word = chr(ord(word)+1)
            
            if word == '{':
                word = 'a'
            
            if word in skip_words:
                continue
            else:
                tmp_index -= 1
        
        answer += word
            
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        
        for(int i=0; i<s.length(); i++) {
            char ch = s.charAt(i);
            int tmp_index = index;
            while(tmp_index > 0) {
                ch += 1;
                
                if(ch > 'z') {
                    ch -= 26; // z 넘어가면 다시 a로
                }
                
                if(skip.contains(String.valueOf(ch))) {
                    continue;
                } else {
                    tmp_index--;
                }
            }
            
            answer += ch;
        }
        
        return answer;
    }
}